syncano/models/manager.py

```diff
@@ -173,7 +193,30 @@ def bulk_create(self, *objects):
         .. warning::
             This method is not meant to be used with large data sets.
         """
-        return [self.create(**o) for o in objects]
+
+        if len(objects) > 50:
+            raise SyncanoValueError('Only 50 objects can be created at once.')
+
+        class_name = objects[0].get('class_name')
+        for o in objects[1:]:
+            next_class_name = o.get('class_name')
+            if o.get('class_name') != class_name:
+                raise SyncanoValidationError('Bulk create can handle only objects of the same type.')
```

Comment:
Ok, let's keep it this way then. In Django it is more of a technical limitation than a readability improvement as INSERT into multiple tables in bulk doesn't make sense as it could end up just as bad as 1 insert per table. In our case, we could still improve it as you noted somewhere along the road but for now let's keep it simple.